default/generated/azure/31-openliberty-jms.yaml

```diff
@@ -0,0 +1,235 @@
+  description: Detect JMS resource references in web.xml/ejb-jar.xml
+  effort: 3
+  labels:
+  - konveyor.io/source=java-ee
+  - konveyor.io/target=azure-aks
+  - domain=azure-readiness
+  - category=openliberty-migration
+  - openliberty
+  message: |
+    JMS resource references in deployment descriptors detected that require removal and replacement with Azure Service Bus configuration for AKS deployment.
+
+    **Migration Requirements:**
+    • **Remove JMS Configuration:** Clean up resource-ref entries from web.xml and ejb-jar.xml
+    • **Externalize Configuration:** Move messaging configuration to application properties and environment variables
+    • **Update Resource Management:** Replace container-managed JMS resources with application-managed Azure clients
+    • **Security Updates:** Migrate from container authentication to Azure Service Bus authentication
+
+    **Configuration Migration Steps:**
+    1. **Identify JMS Dependencies:** Catalog all JMS resource references in deployment descriptors
+    2. **Remove XML Configuration:** Delete resource-ref, message-driven-destination elements
+    3. **Add Azure Properties:** Configure Azure Service Bus connection parameters
+    4. **Update Code References:** Replace JNDI lookups with direct Azure client usage
+    5. **Validate Configuration:** Test connection and message operations
+
+  links:
+  - title: "Azure Service Bus Java configuration"
+    url: "https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-java-how-to-use-queues"
+  - title: "Spring Boot externalized configuration"
+    url: "https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config"
+  - title: "Kubernetes ConfigMaps and Secrets"
+    url: "https://kubernetes.io/docs/concepts/configuration/"
+  ruleID: openliberty-jms-00004
+  when:
+    builtin.xml:
+      filepaths:
+      - web.xml
+      - ejb-jar.xml
+      namespaces:
+        jee: http://xmlns.jcp.org/xml/ns/javaee
+        j2ee: http://java.sun.com/xml/ns/j2ee
+      xpath: //jee:resource-ref/jee:res-type[contains(text(), 'javax.jms')] | //jee:message-driven-destination | //j2ee:resource-ref/j2ee:res-type[contains(text(), 'javax.jms')] | //j2ee:message-driven-destination
+
+- category: mandatory
+  customVariables: []
+  description: Detect JMS JNDI lookups in code
+  effort: 4
+  labels:
+  - konveyor.io/source=java-ee
+  - konveyor.io/target=azure-aks
+  - domain=azure-readiness
+  - category=openliberty-migration
+  - openliberty
+  message: |
+    Detected manual JNDI lookups for JMS resources that need to be updated for Azure messaging services.
+    
+    **Required changes:**
+    - Remove JNDI lookups for JMS resources
+    - Replace with direct Azure Service Bus client initialization
+    - Update message sending and receiving logic
+  ruleID: openliberty-jms-00006
+  when:
+    builtin.filecontent:
+      filePattern: .*\.java
+      pattern: \.lookup\(\s*["'][^"']*jms[^"']*["']\s*\)
```

Comment:
is it duplicated with openliberty-jms-00002?
E.g.
```
@Resource(lookup = "java:/ConnectionFactory")
```
two rule match for one line? maybe we can merge them into one?